app/routes/_protected.tsx

```diff
@@ -1,17 +1,39 @@
-import { Outlet, useLocation } from '@remix-run/react';
+import type { LoaderFunction } from '@remix-run/node';
+import { Outlet, useLoaderData, useLocation } from '@remix-run/react';
 import { Redirect } from '~/components/redirect';
 import { LoadingScreen } from '~/features/loading/LoadingScreen';
 import { useAuthState } from '~/features/user/auth';
 import { shouldVerifyEmail as shouldUserVerifyEmail } from '~/features/user/user';
 import { UserProvider } from '~/features/user/user-provider';
+import { type ExchangeRates, fetchRates } from '~/lib/exchange-rate';
 
+export const loader: LoaderFunction = async (): Promise<{
+  rates: ExchangeRates;
+}> => {
+  const rateSource = 'average';
+  try {
+    const rates = await fetchRates(rateSource);
```

Comment:
I think we should use react query instead. I'll make an example of how I think it should look

Reply:
See here: https://github.com/MakePrisms/boardwalkcash/pull/225

Reply:
it also solves the prop drilling issue that you mentioned